build-tools/tasks/test-utils.js

```diff
@@ -22,16 +22,22 @@ function compileTypescript(theme) {
 
 function generateTestUtilsForComponents(signalCompletion) {
   const componentNamesKebabCase = listPublicItems(path.join(testUtilsSrcDir, 'dom'));
+  const publicComponents = listPublicItems('src');
 
-  const components = componentNamesKebabCase.map(testUtilsFolderName => {
-    const name = pascalCase(testUtilsFolderName);
-    const pluralName = pluralizeComponentName(name);
-    return {
-      name,
-      pluralName,
-      testUtilsFolderName,
-    };
-  });
+  const isPublicComponent = name => publicComponents.includes(name);
+
+  const components = componentNamesKebabCase
+    .filter(component => isPublicComponent(component) || component === 'annotation')
```

Comment:
How about silently skipping also throw an error that there is a folder in an unexpected location?

Because there are also some teams importing these directories directly assuming "if they are not in `internal`, it must be public"

Reply:
Okay, I found we have a snapshot test: https://github.com/cloudscape-design/components/blob/main/src/__tests__/snapshot-tests/__snapshots__/test-utils-wrappers.test.tsx.snap

So this is already validated if anything unexpected is exposed

Reply:
(feel free to resolve the thread)